gnat2goto/driver/tree_walk.adb

```diff
@@ -3107,8 +3242,12 @@ package body Tree_Walk is
 
    function Do_Type_Definition (N : Node_Id; Discs : List_Id) return Irep is
    begin
-      pragma Assert (Discs = List_Id (Types.Empty)
-                       or else Nkind (N) = N_Record_Definition);
+      if Discs /= List_Id (Types.Empty)
+        or else Nkind (N) = N_Record_Definition
```

Comment:
This should be `Discs /= List_Id (Types.Empty) and then Nkind (N) /= N_Record_Definition`

Reply:
Fixed, thanks.